File: NumberFilter.java
public class NumberFilter {
    private IntPredicate intPredicate;
 
    public void setIntPredicate(IntPredicate intPredicate) {
        this.intPredicate = intPredicate;
    }
 
    public void accept(int i) {
        if(intPredicate.accept(i))
          Log.d("NUMBERFILTER", "Number accepted");
        else
           Log.d("NUMBERFILTER", "Number not accepted");
    }
}

File: NumberFilter.kt
class NumberFilter {
  
    var intPredicate:(i:Int)->Boolean = {it >= 0} // Default implementation
                                   
    fun accept(i:Int) {
        if(intPredicate(i))
            Log.d("NUMBERFILTER", "Number accepted");
        else
            Log.d("NUMBERFILTER", "Number not accepted");
    }
}